plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/util/FeatureDevService.kt

```diff
@@ -135,9 +135,11 @@ class FeatureDevService(val proxyClient: FeatureDevClient, val project: Project)
                 logger.warn(e) { "StartTaskAssistCodeGeneration failed for request: ${e.requestId()}" }
 
                 if (e is software.amazon.awssdk.services.codewhispererruntime.model.ServiceQuotaExceededException || (
-                        e is software.amazon.awssdk.services.codewhispererruntime.model.ThrottlingException && e.message?.contains(
-                            "limit for number of iterations on a code generation"
-                        ) == true
+                        e is software.amazon.awssdk.services.codewhispererruntime.model.ThrottlingException && (
+                            e.message?.contains(
+                                "limit for number of iterations on a code generation"
+                            ) == true || e.message?.contains("StartTaskAssistCodeGeneration reached for this month.") == true
+                            )
                         )
```

Comment:
can we have a test for all the cases that mean the same thing

Reply:
Yes, I will add now